mono/profiler/mono-profiler-log.c

```diff
@@ -1335,7 +1392,7 @@ send_log_unsafe (gboolean if_needed)
 static void
 sync_point_flush (void)
 {
-	g_assert (InterlockedReadPointer (&buffer_rwlock_exclusive) == (gpointer) thread_id () && "Why don't we hold the exclusive lock?");
+	g_assert (InterlockedRead (&buffer_lock_state) == PROF_TLS_GET ()->small_id << 16 && "Why don't we hold the exclusive lock?");
```

Comment:
Good point, and it probably wouldn't hurt to have some macros for accessing/setting the lock state either.